### PR TITLE
remove deprecated framework.ExpectNotEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Ref #115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
unrelated failures 
/test pull-kubernetes-e2e-gce

**Comment by danwinship** [comment]
/approve
for the sig-network parts

**Review by SataQiu — COMMENTED** [review]
/lgtm

File: test/e2e/framework/expect.go [hunk 1]
```diff
@@ -293,13 +293,6 @@ func (f *FailureError) backtrace() {
 //	}
 var ErrFailure error = FailureError{}
 
-// ExpectNotEqual expects the specified two are not the same, otherwise an exception raises
-//
-// Deprecated: use gomega.Expect().ToNot(gomega.Equal())
-func ExpectNotEqual(actual interface{}, extra interface{}, explain ...interface{}) {
-	gomega.ExpectWithOffset(1, actual).NotTo(gomega.Equal(extra), explain...)
-}
```
Comment (pohly) [inline test/e2e/framework/expect.go:301]: Yeah! :smile:

**Review by pohly — APPROVED** [review]
/approve

**Pull request merged**